src/main/java/graphql/util/TraverserContext.java

```diff
@@ -0,0 +1,80 @@
+package graphql.util;
+
+import graphql.Internal;
+
+import java.util.Set;
+
+/**
+ * Traversal context
+ *
+ * @param <T> type of tree node
```

Comment:
If we expect **T** of a certain type, would it be better to define upper bound as `T extends Node`?

By the way, I see visitor+traverser context as a specialized mechanism to traverse AST rather than general-purpose visitor.

Reply:
TraverserContext is used by the general Traverser, so I don't really want to  bound it to the Ast `Node`: I guess we can also use it to traverse a `GraphQLSchema` or other tree structures. I think one of you mentioned this also before. Or did I get this wrong?